test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java

```diff
@@ -1471,9 +1479,80 @@ public void testGetDocContentSiteCollection() throws Exception {
         .build()), response.getNamedResources());
     assertEquals(URI.create("http://localhost:1/sites/SiteCollection"),
         response.getDisplayUrl());
-    assertEquals(goldenGroups, pusher.getGroups());
+    assertEquals(goldenGroups, pusher.getGroupDefinitions());
   }
   
+  @Test
+  public void testGetDocContentSCVerifyGroupPush() throws Exception {
+    final Map<GroupPrincipal, Collection<Principal>> goldenGroups1;
+    {
+      Map<GroupPrincipal, Collection<Principal>> tmp
+          = new TreeMap<GroupPrincipal, Collection<Principal>>();
+      tmp.put(SITES_SITECOLLECTION_OWNERS, Arrays.<Principal>asList(
+          GDC_PSL_ADMINISTRATOR));
+      tmp.put(SITES_SITECOLLECTION_MEMBERS, Arrays.asList(
+            new UserPrincipal("GDC-PSL\\spuser2", DEFAULT_NAMESPACE),
+            new GroupPrincipal("BUILTIN\\users", DEFAULT_NAMESPACE),
+            new UserPrincipal("GDC-PSL\\spuser4", DEFAULT_NAMESPACE)));
+      tmp.put(SITES_SITECOLLECTION_VISITORS, Arrays.<Principal>asList());
+      goldenGroups1 = Collections.unmodifiableMap(tmp);
+    }
+
+    final Map<GroupPrincipal, Collection<Principal>> goldenGroups2;
+    {
+      Map<GroupPrincipal, Collection<Principal>> tmp
+          = new TreeMap<GroupPrincipal, Collection<Principal>>();
+      tmp.put(SITES_SITECOLLECTION_OWNERS, Arrays.<Principal>asList(
+          GDC_PSL_ADMINISTRATOR));
+      tmp.put(SITES_SITECOLLECTION_MEMBERS, Arrays.asList(
+            new GroupPrincipal("BUILTIN\\users", DEFAULT_NAMESPACE),
+            new UserPrincipal("GDC-PSL\\spuser4", DEFAULT_NAMESPACE)));
+      tmp.put(SITES_SITECOLLECTION_REVIEWERS, Arrays.asList(
+            new UserPrincipal("GDC-PSL\\spuser2", DEFAULT_NAMESPACE),
+            new GroupPrincipal("BUILTIN\\users", DEFAULT_NAMESPACE)));
+      goldenGroups2 = Collections.unmodifiableMap(tmp);
+    }
+
+    ContentExchange scContentExchange1 = new ContentExchange(
+        ObjectType.SITE_COLLECTION, null, null, null, true, false, null,
+        loadTestString("sites-SiteCollection-sc.xml"), true);
+    ContentExchange scContentExchange2 = new ContentExchange(
+        ObjectType.SITE_COLLECTION, null, null, null, true, false, null,
+        loadTestString("sites-SiteCollection-sc-useonce.xml"), false);
```

Comment:
Maybe flip these? You're using the existing xml with useOnce = true, then the new useonce.xml with useOnce = false?

Reply:
Done.